src/service/workers/utils/utils.go

```diff
@@ -87,3 +87,11 @@ func Convertto18Decimals(amount string) string {
 	ret := big.NewInt(0).Mul(big.NewInt(1000000000000), value)
 	return ret.Text(10)
 }
+
+func ConvertDecimals(originDecimals, destDecimals uint8, amount string) string {
+	origin := new(big.Int).SetInt64(int64(math.Pow10(int(originDecimals))))
```

Comment:
Check please `originDecimals` and `destDecimals` are > 0 and <64 in other case you will have `float64` conversion to the `int64` with udefined result.